engine/table/src/main/java/io/deephaven/engine/util/PythonDeephavenSession.java

```diff
@@ -77,8 +78,14 @@ public PythonDeephavenSession(
             @Nullable final Listener listener, boolean runInitScripts, boolean isDefaultScriptSession)
             throws IOException {
         super(listener, isDefaultScriptSession);
-
-        JpyInit.init(log);
+        try {
+            JpyInit.init(log);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
```

Comment:
This just sets a landmine for other code.

Reply:
Most code that does not _expect_ an interrupt  should either 1) rethrow the interrupted exception, or 2) mark the thread as interupted and return, or mark and throw an unchecked exception. Failing to do that, higher level code up the stack won't get the control they need - it's usually logic that grpc, thread pools, or thread-aware code is responsible for. Unless the piece of code is specifically dealing with a concurrency regime where it knows how to handle the interupts itself, this is the pattern to use. See article by Brian Goetz

https://itblackbelt.wordpress.com/2006/05/24/dealing-with-interruptedexception-by-brian-goetz/

Don’t swallow interrupts

Sometimes throwing InterruptedException is not an option, such as when a task defined by Runnable calls an interruptible method. In this case, you can’t rethrow InterruptedException, but you also do not want to do nothing. When a blocking method detects interruption and throws InterruptedException, it clears the interrupted status. If you catch InterruptedException but cannot rethrow it, you should preserve evidence that the interruption occurred so that code higher up on the call stack can learn of the interruption and respond to it if it wants to. This task is accomplished by calling interrupt() to “reinterrupt” the current thread, as shown in Listing 3. At the very least, whenever you catch InterruptedException and don’t rethrow it, reinterrupt the current thread before returning.

Reply:
Yeah I'm firmly with Devin on this, though we could change the contract for script sessions to throw InterruptedException as another option, so it is clear that any session impl could do this.

Reply:
There are really only two possibilities, here:
1. The higher level code expects an interrupt might happen. In that case, re-throwing InterruptedException makes the most sense.
2. The higher level code doesn't. It will never check the interrupt status, leaving it set. This is what I mean about "a landmine" - completely unrelated code that expects interruption might see the interrupted status left over in dirty state.

If you're planning to let the thread die anyway, it doesn't matter much. Otherwise, you aren't doing anyone any favors. If you think this should be handled by calling code, re-throw it.

Reply:
I do see the point about thread pools, etc. My major concern is "any random thread we control". If we expect to outlive the exception, it's a liability to have that flag set.

Reply:
Arguably, we may want to change how startup occurs, so the session constructor isn't responsible for starting jpy.

Reply:
Regardless, we don't expect to outlive the exception, right? At that point, this is just a moot discussion. The code can stay as-is.